apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -241,16 +259,33 @@ export const submitChatContents = createAsyncThunk(
       status: Status.OK,
       chatMessageText: newUserMessageText,
       timestamp: getCurrentTimestamp(),
+      sessionId: currentSessionId,
     };
     thunkAPI.dispatch(addChatMessage(newMessage));
 
     // Post user content and messages to backend and retrieve assistant response.
+
     const chatApiResponse = await postAichatCompletionMessage(
       newUserMessageText,
-      storedMessages,
+      currentSessionId
+        ? storedMessages.filter(
+            message => message.sessionId === currentSessionId
+          )
+        : [],
       aiCustomizations,
       aichatContext
     );
+    console.log('chatApiResponse', chatApiResponse);
+
+    // TODO: error handling
+    thunkAPI.dispatch(setChatSessionId(chatApiResponse.sessionId));
+    thunkAPI.dispatch(
+      updateChatMessageSession({
+        id: newMessage.id,
+        sessionId: chatApiResponse.sessionId,
```

Comment:
Discussed offline - an alternate approach here could be to treat the list of messages & updates in the chat window as more of a generic timeline of events, and just push an internal-only "new chat session" event whenever we reset the chat session. That way we don't need to go back and manually update the first message whenever the chat session refreshes. We could even store the associated session ID with that new chat session event, and just grab it whenever we need to send it to the server. I'd be okay either way but curious what you think after spending some time with this?

Reply:
hmm yeah honestly I'm torn on this stuff...I think one downside to storing the session ID with the new chat session event would be that when we want to wipe it (eg, when a model customization changes), we'd have to go find that message and update it.

I guess having iterated on this a bit, it feels cleaner to me to rely on explicit session IDs per message, rather than ordering of messages. Deciding which messages to send to the server is pretty clean as well (if there's no session ID, send nothing. if there is one, send all matching messages). And having to go back and update the last message doesn't feel particularly gross either, because we already have the ID of the message to update on hand once the request comes back from the server.